doc/versioning_and_life_cycle.md

```diff
@@ -0,0 +1,72 @@
+
+- The **GA** state means that a feature is mature, stable, available to all customers, and has formal support. A GA feature must be forward compatible and not prone to cause breaking changes.
+    - Commands not tagged with the `is_preview` flag are considered as GA commands
+
+- The **Preview** state means a feature is unstable and not finalized. Preview features are released before GA and allow customers to test, experiment, and provide feedback in advance of the GA release. Preview features should not be used in production environments as they aren't stable or finalized and could have breaking changes with each release up to, and including, the feature's GA release.
+    - Preview features and commands need to be tagged with the [is_preview](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#preview-commands-and-arguments) flag to let customers know about instability.
+    - It is generally recommended to put the preview features into CLI extension first and migrate it to core CLI after they are stable and GA. We can also consider moving some preview features from core CLI to CLI extensions if they are proper for moving.
+
+### States Transition
+
+#### Preview to GA
+Generally, if the service is stable and the design of the commands will not change, we are confident that these commands will not cause breaking changes in the future, then we can remove the `is_preview` flag and GA those commands.
+
+#### GA to Preview
+For the core CLI, it is generally not allowed to downgrade from GA to Preview, because the GA state means that the stability of commands has been promised to customers, and new instability cannot be introduced by downgrading to Preview. Moreover, since Azure CLI is an all-in-one package, users usually upgrade directly to the latest version of CLI. If users skip the intermediate version with features having been downgraded to Preview and directly upgrade to the latest version that has breaking changes, it will make a confusing and frustrating experience for them.
+
+### Relationship with api-version
+
+At present, CLI sets the api-version according to the Resource Provider granularity, and there are corresponding configurations for different profiles. For most commands, we do not support customers to specify the requested api-version.
+
+- For the GA api-version, if the command itself determines the final design, then we usually make the implemented command also in the GA state.
+
+- For the public preview api-version, the situation is relatively complicated, and we usually divide it into the following cases:
+    - if those features are newly onboard RPs, command groups or extensions, we usually mark them as `is_preview`. 
```

Comment:
Please consider numbering the cases to distinguish them from the bullet points above and below (don't rely solely on the indent). 
```suggestion
    1. if those features are newly onboard RPs, command groups or extensions, we usually mark them as `is_preview`. 
```